File: fingerprint-sdk/fingerprint_sdk/lib/src/models/sdk_error.dart
/// SDK error codes
enum SdkErrorCode {
  cameraPermissionDenied,
  cameraInitFailed,
  lowQuality,
  livenessFailed,
  timeout,
  deviceUnsupported,
  processingError,
  captureAborted,
  unknown,
}

class SdkException implements Exception {
  final SdkErrorCode code;
  final String message;

  const SdkException(this.code, this.message);

  factory SdkException.fromMap(Map<dynamic, dynamic> m) {
    final code = _parseCode(m['errorCode'] as String? ?? 'UNKNOWN');
    return SdkException(code, m['errorMessage'] as String? ?? 'Unknown error');
  }

  static SdkErrorCode _parseCode(String c) => switch (c) {
        'CAMERA_PERMISSION_DENIED' => SdkErrorCode.cameraPermissionDenied,
        'CAMERA_INIT_FAILED'       => SdkErrorCode.cameraInitFailed,
        'LOW_QUALITY'              => SdkErrorCode.lowQuality,
        'LIVENESS_FAILED'          => SdkErrorCode.livenessFailed,
        'TIMEOUT'                  => SdkErrorCode.timeout,
        'DEVICE_UNSUPPORTED'       => SdkErrorCode.deviceUnsupported,
        'PROCESSING_ERROR'         => SdkErrorCode.processingError,
        'CAPTURE_ABORTED'          => SdkErrorCode.captureAborted,
        _                          => SdkErrorCode.unknown,
      };

  @override
  String toString() => 'SdkException(${code.name}): $message';
}

File: fingerprint-sdk/fingerprint_sdk/ios/Classes/Bridge/SDKError.swift
import Foundation

enum SDKError: Error {
    case cameraPermissionDenied(String)
    case deviceUnsupported(String)
    case timeout(String)
    case lowQuality(String)
    case livenessFailed(String)
    case captureAborted(String)
    case unknown(String)

    var code: String {
        switch self {
        case .cameraPermissionDenied: return "CAMERA_PERMISSION_DENIED"
        case .deviceUnsupported: return "DEVICE_UNSUPPORTED"
        case .timeout: return "TIMEOUT"
        case .lowQuality: return "LOW_QUALITY"
        case .livenessFailed: return "LIVENESS_FAILED"
        case .captureAborted: return "CAPTURE_ABORTED"
        case .unknown: return "UNKNOWN"
        }
    }

    var message: String {
        switch self {
        case .cameraPermissionDenied(let m), .deviceUnsupported(let m),
             .timeout(let m), .lowQuality(let m), .livenessFailed(let m),
             .captureAborted(let m), .unknown(let m): return m
        }
    }
}
